validator/src/network/rest/mod.rs

```diff
@@ -1,29 +1,59 @@
+use crate::services::mixmining;
 use iron::prelude::*;
+use presence::mixnode;
+use presence::topology;
 use router::Router;
+use std::sync::{Arc, Mutex};
```

Comment:
I am blocking way more than I need to anyway - I'm planning to move the mutex from the Service (where it is now) to the level of each individual datastore (when those exist), because right now every request is basically going to hit the mutex for every other request. I'll switch mutexes when I do that next bit of work (right now I only have 1 datastore so it doesn't really make much difference) - if you happen to have the source on the futures vs std mutex please shoot it to me in Keybase, I'm interested to see!